src/imageio/format/tiff.c

```diff
@@ -165,67 +165,85 @@ int write_image(dt_imageio_module_data_t *d_tmp, const char *filename, const voi
    As there might be pipeline errors at the border we leave them alone.
    After these checks layers can be used later on.
 */
-  uint16_t layers = 3;  // default are rgb images
+  volatile uint16_t layers = 3;  // default are rgb images
 
-  if((d->global.height > 4) && (d->global.width > 4) && d->shortfile)
+  if(d->shortfile && (d->global.height > 4) && (d->global.width > 4))
   {
     layers = 1;    // let's now assume a grayscale
     if(d->bpp == 32 || (d->bpp == 16 && d->pixelformat))
     {
-      for(int y = 1; y < d->global.height-1; y++)
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(in_void, d) \
+  shared(layers) \
+  schedule(simd:static) \
```

Comment:
`static` means that the number of loop iterations (if computable at loop start) will be evenly divided among the available threads, after which no further parallelization work is needed.  `dynamic` means that a small number of loops iterations will be assigned to each thread initially, and each will then assign itself additional iterations on finishing a chunk.  This has more overhead (especially since there is a single memory location updated by all threads), but can be faster if individual loop iterations vary wildly in execution time.  There aren't really any loops in dt where that is the case.

Endianness won't affect performance here.  P/E cores will have some effect, since iterations done on E cores will take longer than those done on P cores, but might not outweigh the extra overhead of `dynamic` and I don't know of any way to switch the scheduling at runtime.  We could add a preprocessor macro to select static versus dynamic at compile time if that does prove beneficial for P/E machines.

Reply:
I didn't mean endian but bug/little in ARM speak.